internal/controller/postgrescluster/instance.go

```diff
@@ -1270,8 +1270,19 @@ func generateInstanceStatefulSetIntent(_ context.Context,
 	// - https://docs.k8s.io/tasks/configure-pod-container/share-process-namespace/
 	sts.Spec.Template.Spec.ShareProcessNamespace = initialize.Bool(true)
 
+	// Patroni calls the Kubernetes API and pgBackRest may interact with a cloud
+	// storage provider. Use the instance ServiceAccount and automatically mount
+	// its Kubernetes credentials.
+	// - https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity
+	// - https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
 	sts.Spec.Template.Spec.ServiceAccountName = instanceServiceAccountName
 
+	// Disable environment variables for services other than the Kubernetes API.
+	// - https://docs.k8s.io/concepts/services-networking/connect-applications-service/#accessing-the-service
+	// - https://releases.k8s.io/v1.16.0/pkg/kubelet/kubelet_pods.go#L523-L533
+	// - https://releases.k8s.io/v1.23.0/pkg/kubelet/kubelet_pods.go#L553-L563
```

Comment:
:thinking: I see the two links above are from different releases, but they are otherwise the same. Why do we need both? Is it related to the surrounding code?

Reply:
Ah, I see. Maybe just a note that this field has been available since 1.13 or similar.

To the second find, I'm not sure if that's worth documenting here as it isn't really related to the update, but I suppose it couldn't hurt to have it either.

Reply:
I've removed the link to v1.16 and mentioned v1.13 in the commit message.